File: common/src/main/java/dev/mayaqq/estrogen/config/ChestConfig.java
package dev.mayaqq.estrogen.config;

import com.teamresourceful.bytecodecs.base.ByteCodec;
import com.teamresourceful.bytecodecs.base.object.ObjectByteCodec;

public record ChestConfig(boolean enabled, boolean armorEnabled, boolean physicsEnabled, float bounciness, float damping) {

    public static final ByteCodec<ChestConfig> BYTE_CODEC = ObjectByteCodec.create(
            ByteCodec.BOOLEAN.fieldOf(ChestConfig::enabled),
            ByteCodec.BOOLEAN.fieldOf(ChestConfig::armorEnabled),
            ByteCodec.BOOLEAN.fieldOf(ChestConfig::physicsEnabled),
            ByteCodec.FLOAT.fieldOf(ChestConfig::bounciness),
            ByteCodec.FLOAT.fieldOf(ChestConfig::damping),
            ChestConfig::new
    );
}

File: src/main/kotlin/dev/mayaqq/estrogen/config/types/ChestConfig.kt
package dev.mayaqq.estrogen.config.types

import kotlinx.serialization.Serializable

@Serializable
data class ChestConfig(
    val enabled: Boolean,
    val armorEnabled: Boolean,
    val physicsEnabled: Boolean,
    val bounciness: Float,
    val damping: Float
)

